File: python/387 First Unique Character in a String.py
class Solution(object):
    def firstUniqChar(self, s):
        if len(s) == 0:
            return -1
        dic = {}
        for i in s:
            if i in dic:
                dic[i] += 1
            else:
                dic[i] = 1
        for j in range(len(s)):
            if dic[s[j]] == 1:
                return j
        return -1

File: java/387 First Unique Character in a String.java
class Solution {
    public int firstUniqChar(String s) {
        int[] tmp = new int[26];
        int ans = -1;
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            ++tmp[c - 'a'];
        }

        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            if (tmp[c-'a'] == 1) {
                ans = i;
                break;
            }
        }
        return ans;
    }
}
